### ISSUE TITLE
Pull Request Reminders

### ISSUE THREAD

**Issue by arunpkqt** [integrations/slack#789]
It would be great If GitHub app can send reminders to the channel if there are Open Pull Requests which are still to be reviewed. There should be an option using which users can configure the interval for the reminders or remind the channel about all open PRs when a new PR is created or any existing PR is updated/closed.

**Comment by neb** [comment]
@arunpkqt See also https://github.com/integrations/slack/issues/380 from a year ago, including the commercial product mentioned in the comments. (I am not affiliated, but it is a pretty good implementation of exactly what you requested! And not too expensive.)

**Comment by arunpkqt** [comment]
Thanks, @neb I have seen this but didn't give it a try. Will check.

**Comment by abinoda** [comment]
@neb Thanks for the shout-out! I maintain [Pull Reminders](https://github.com/marketplace/pull-reminders/) 👋 

If pricing is a problem for anyone reading this, please email me at [abi@pullreminders.com](mailto:abi@pullreminders.com) and I'd be happy to give your team a free account.

<img width="480" src="https://user-images.githubusercontent.com/50083/53290403-a2d88e00-3760-11e9-9cde-58ee70344a37.png">

**Comment by Dunedan** [comment]
Yes it is still relevant. Pull Reminders got purchased by Github and creating new integrations with Pull Reminders doesn't work anymore (I always get some kind of redirect loop when trying to do so). On the other hand the Github Slack integration afaik doesn't support similar functionality yet.

**Comment by radix** [comment]
@pullreminders was acquired by github over 8 months ago, disabled signups, and there is still nothing to replace it. @abinoda you still handing out accounts?

**Comment by JesusGVideoAmp** [comment]
I'm also looking for alternatives here @abinoda

**Comment by orj** [comment]
GitHub has docs on setting up Pull Request Reminders but I can't find the option in my Company's Team settings.

https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-scheduled-reminders-for-pull-requests

**Comment by moraesdam** [comment]
Followers, please check here [#380](https://github.com/integrations/slack/issues/380)

**Comment by orj** [comment]
As an update I figured out that you need opt in to the beta for pull reminders.

**Comment by doublejosh** [comment]
Another option https://github.com/ekmartin/slack-pull-reminder

**Comment by arunpkqt** [comment]
`Scheduled Reminders` are now available from GitHub. More info can be found here
https://github.blog/2020-04-21-stay-on-top-of-your-code-reviews-with-scheduled-reminders/

**Comment by tommeier** [comment]
Don't migrate to the new one yet - they don't seem to have a mapping of slack username to github username configurable anywhere (I regret migrating).

**Comment by chenbobby** [comment]
@tommeier What's your use case? Do you have multiple slack username/workspaces and multiple github usernames?

I haven't used the new Scheduled Reminders yet but I'm guessing that they only support 1 github to 1 slack user...?

**Comment by tommeier** [comment]
@chenbobby we do have multiple - but this is scoped to one workspace - the current migrated implementation doesn't support mapping at all AFAIK.

pullreminders.com had a fantastic list of github users - option to add - then map to the slack username. Great, simple implementation.
![image](https://user-images.githubusercontent.com/19973/84443752-9d6bc380-ac83-11ea-80db-7ffd057d3c15.png)

Our use case is simply that - a mapping of github users to _one_ slack username for the current workspace the reminders are set to post in. I, wrongly, assumed that migrating to github reminders it would migrate this mapping. We'd become pretty reliant on pullreminders.com as it pinged everyone concerned and everyone cleared the deck before commencing their work day.

Choosing 'Manage reminder' takes you to https://github.com/organizations/XXXXX/settings/reminders/XXXX - which has a bunch of settings - but no github username to slack mapping.

There is some _automatic_ mapping - I'm guessing based on username or slack 'friendly name' - which works _some_ of the time. With around ~50 developers - i'm seeing about _10%_ of the notifications mapping to the slack username correctly. Even developers that map their _exact_ github username as their slack name - no mapping.

Basically its near-useless without that mapping for us. Which is such a shame - it was _fantastic_ before migrating.

**Comment by chenbobby** [comment]
@tommeier I think I'm getting what you're saying. It would definitely useful to map github users to slack usernames. This seems to a be a hot issue, according to #749 and #784 (also about linking github users to slack usernames).

I did some digging and found the `/signin` command, but it didnt seem to change anything in my bot notifications.

![Screen Shot 2020-06-11 at 10 13 06 PM](https://user-images.githubusercontent.com/14797301/84467359-c1ea8000-ac30-11ea-8135-f18d5b321a60.png)

It's exactly what you said:

> Even developers that map their exact github username as their slack name - no mapping.

Maybe Github's "Scheduled Reminders" will get some updates soon for better integration.

btw, I'm also curious to know how PullReminders.com did their team organizations. Was the PullReminders.com account centered around a single Slack workspace, or around a single Github organization? Did people use PullReminders.com on their personal Github accounts, without an organization?

**Comment by slmingol** [comment]
I'd still like to see this feature integrated into the slack app. The scheduled reminders feature for our instances of repos in GitHub Enterprise have the Settings section of each team disabled for obvious reasons. By doing so it cuts you off from that feature completely.

- https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-scheduled-reminders-for-your-team#about-scheduled-reminders-for-teams

![image](https://user-images.githubusercontent.com/15072/88337306-84374780-cd04-11ea-8e9e-d775db57e42a.png)

**Comment by greveritt** [comment]
I've been trying to get direct message reminders of review requests sent to a couple multi-channels guests in Slack. They were able to get such messages from Pull Reminders, but have not been able to get reminder messages since I migrated us to Scheduled Reminders. Can a full member of the Slack workspace set up the Slack integration to message the guests when needed?

The [Slack documentation](https://slack.com/help/articles/202518103-Multi-Channel-and-Single-Channel-Guests#guest-permissions) says that guests cannot connect to Slack integrations by themselves. Whenever our guests try to add scheduled reminders of their own, they end up seeing an error saying that Slack won't let them "install" the GitHub app to our Slack workspace, even though I already set up the Slack app myself and they have no need to "install" any app at all.

Thanks for listening to my concerns.

![step 1](https://user-images.githubusercontent.com/1747920/88589963-f0c58580-d027-11ea-9c24-7fb636e994b5.png)

![step 2](https://user-images.githubusercontent.com/1747920/88589975-f622d000-d027-11ea-8280-e4364f5f3864.png)

**Comment by nabroyan** [comment]
Any update on Github username to Slack username mapping?

**Comment by tatane616** [comment]
https://github.com/integrations/slack/issues/789#issuecomment-664627329 
I'm having the same problem, has anyone solved this?

**Comment by jibsonline** [comment]
+1

**Comment by rogeryen** [comment]
Also migrated over to scheduled reminders only to find out that it has stopped working completely for me. Sending a test message also doesn't work. Does this feature actually work for anyone?

**Comment by jibsonline** [comment]
@rogeryen that could be because the GitHub bot is not enabled on the channel you are trying to push notifications

**Comment by rogeryen** [comment]
> @rogeryen that could be because the GitHub bot is not enabled on the channel you are trying to push notifications

Does it only work in channels? Because I'm really only interested in real time notifications, which I interpreted as DM from the `@github` bot?

**Comment by jibsonline** [comment]
> Does it only work in channels? Because I'm really only interested in real time notifications, which I interpreted as DM from the `@github` bot? 
Looking at the placeholder seems like it only works on channels
![image](https://user-images.githubusercontent.com/1730931/105132210-84c46580-5b10-11eb-849e-2c7593e2e957.png)

You need to add `@GitHub` to the channel

**Comment by rogeryen** [comment]
> > Does it only work in channels? Because I'm really only interested in real time notifications, which I interpreted as DM from the `@github` bot?
> 
> Looking at the placeholder seems like it only works on channels
> ![image](https://user-images.githubusercontent.com/1730931/105132210-84c46580-5b10-11eb-849e-2c7593e2e957.png)
> 
> You need to add `@GitHub` to the channel

Sucks if that's the case because pullreminders was working perfectly for me and I had to migrate myself over...

**Comment by rogeryen** [comment]
> > > Does it only work in channels? Because I'm really only interested in real time notifications, which I interpreted as DM from the `@github` bot?
> > 
> > 
> > Looking at the placeholder seems like it only works on channels
> > ![image](https://user-images.githubusercontent.com/1730931/105132210-84c46580-5b10-11eb-849e-2c7593e2e957.png)
> > You need to add `@GitHub` to the channel
> 
> Sucks if that's the case because pullreminders was working perfectly for me and I had to migrate myself over...

I did resolve the issue with suggestion from GitHub support to /github signout and /github signin again. 

And also for the record, the “real time notification” feature is indeed DM from the GitHub bot.

**Comment by jibsonline** [comment]
@rogeryen so are you able to send it to yourself or still to a channel only?

**Comment by rogeryen** [comment]
> @rogeryen so are you able to send it to yourself or still to a channel only?

@jibsonline yes I am able to get notifications from the bot to myself only in DM, when I enabled 'real time notification' in my personal settings.

**Comment by anticafe** [comment]
I notice the Pull Request Reminders still notify the approved PRs. 🤔 (for example in this case, it's 2 months old).
Here is my configuration.
![image](https://user-images.githubusercontent.com/340836/144731991-a5101cd9-d2f9-49ea-bd97-bf933a933441.png)

Do you know what's wrong when I check the **Ignore approved pull requests**? I thought it's expected to ignore the approved PRs.

**Comment by IGassmann** [comment]
@anticafe same thing happening on my end. Have you figured how to fix this?

**Comment by cosydney** [comment]
I'm the creator of [axolo]( http://axolo.co/). Axolo creates an ephemeral channel for each PR which is great for remembering the PR you have to review. On the reminder side, we have actionnable reminders who sends a message in this PR channel and tags the right person to act (reviewer, creator) depending on the status of the PR. You can reach out on our website happy to help.

**Comment by rubengmurray** [comment]
After a few false starts I've managed to get the integration working in the sense that it's linked my account and it's posted stale PRs to a Slack channel, but the app is only able to sync with certain repositories despite attempting the 'all repositories' and 'select repositories' options.

I've checked CODEOWNERS in repositories that are working and aren't working and things look identical so I'm not sure why some repos are available and others aren't 🤷 

(Very happy to see the reminders for the ones that work though)
